models/zuora/zuora_subscription.sql

```diff
@@ -1,12 +1,158 @@
+with sub_mrr as (
```

Comment:
@dwallacerjm Dave, this analysis should be in the `analytics/analysis/zuora` folder, not `analytics/models/zuora`. The general idea is to first create base models which map the underlying zuora tables into views that incorporate only what's needed for subsequent transformations and analysis, including all of the casting and any transformation that may be duplicated down the line. In other words: create an abstraction that makes it easier to write and read SQL down the line.

This is exactly what's happening int the original `zuora_subscription.sql`, which is a base model for the underlying `zuora.zuora_subscription` table. There is casting into timestamps and the subscription version is calculated in that view so that we don't have to worry about it down the line. 

Also, let's keep `subscr` instead of `sub` as this is more descriptive to someone going through this analysis in the future.